doc-changelog/action.yml

```diff
@@ -0,0 +1,160 @@
+name: >
+  Documentation Changelog
+
+description: |
+  Adds a newsfragment to the target repository using ``towncrier``
+  containing the title and number of the pull request.
+
+inputs:
+  # Required inputs
+
+  token:
+    description: >
+      GitHub token for doing a git commit & push
```

Comment:
Explicitly say the permissions that the token needs in YML format. Something like:

```yml
permissions:
  content: write
  pull-requests: write
  actions: write
```

I'm not sure which ones are needed - but this needs to be verified with dependabot PRs for example.